### ISSUE TITLE
Crash on startup

### ISSUE THREAD

**Issue by Gregliest** [AppsFlyerSDK/AppsFlyer-Segment-Integration#5]
Getting the following NullPointerException when the app is starting.  It looks like `track` is getting called before any constructor, so the integration does not have a context yet.

```
java.lang.NullPointerException: Attempt to invoke virtual method 'android.content.SharedPreferences android.content.Context.getSharedPreferences(java.lang.String, int)' on a null object reference
    at com.appsflyer.AppsFlyerProperties.getReferrer(AppsFlyerProperties.java:161)
    at com.appsflyer.AppsFlyerLib.trackEvent(AppsFlyerLib.java:520)
    at com.segment.analytics.android.integrations.appsflyer.AppsflyerIntegration.track(AppsflyerIntegration.java:124)
    at com.segment.analytics.IntegrationOperation$10.run(IntegrationOperation.java:185)
    at com.segment.analytics.Analytics.performRun(Analytics.java:1237)
    at com.segment.analytics.Analytics$10.run(Analytics.java:693)
    at android.os.Handler.handleCallback(Handler.java:739)
    at android.os.Handler.dispatchMessage(Handler.java:95)
    at android.os.Looper.loop(Looper.java:145)
    at android.app.ActivityThread.main(ActivityThread.java:5942)
    at java.lang.reflect.Method.invoke(Native Method)
    at java.lang.reflect.Method.invoke(Method.java:372)
    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1399)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1194)
```

**Comment by f2prateek** [comment]
@af-golan @shachar-af This issue is because the context field is only set in [`onActivityCreated`](https://github.com/AppsFlyerSDK/AppsFlyer-Segment-Integration/blob/master/app/src/main/java/com/segment/analytics/android/integrations/appsflyer/AppsflyerIntegration.java#L68) and [`onActivityResumed`](https://github.com/AppsFlyerSDK/AppsFlyer-Segment-Integration/blob/master/app/src/main/java/com/segment/analytics/android/integrations/appsflyer/AppsflyerIntegration.java#L78) (and [`onActivityPaused`](https://github.com/AppsFlyerSDK/AppsFlyer-Segment-Integration/blob/master/app/src/main/java/com/segment/analytics/android/integrations/appsflyer/AppsflyerIntegration.java#L84) for some reason).

Events may be delivered to the AppsFlyer integration before these methods are triggered, e.g. if you're tracking in your Application class or when a non-activity component is active such as a Service or BroadcastReceiver. In these cases, the `context` field will be `null` and throw an Exception originating from [usage of the null context here](https://github.com/AppsFlyerSDK/AppsFlyer-Segment-Integration/blob/master/app/src/main/java/com/segment/analytics/android/integrations/appsflyer/AppsflyerIntegration.java#L124).

Since you're using the application context, you can use the Analytics class to get a reference to it.

```
// https://github.com/AppsFlyerSDK/AppsFlyer-Segment-Integration/blob/master/app/src/main/java/com/segment/analytics/android/integrations/appsflyer/AppsflyerIntegration.java#L48

Context context = analytics.getApplication();
AppsflyerIntegration integration =  new AppsflyerIntegration(context, logger, afLib, devKey);

// https://github.com/AppsFlyerSDK/AppsFlyer-Segment-Integration/blob/master/app/src/main/java/com/segment/analytics/android/integrations/appsflyer/AppsflyerIntegration.java#L58
public AppsflyerIntegration(Context context, Logger logger, AppsFlyerLib afLib, String devKey) {
        this.context = context;
        this.logger = logger;
        this.appsflyer = afLib;
        ...

```

With this, you can also remove the lines I references earlier where the `context` field is being set.

**Comment by shachar-af** [comment]
fixed in v1.3
Thanks @f2prateek
